Public/Src/Cache/ContentStore/Hashing/ContentHasher.cs

```diff
@@ -24,7 +24,10 @@ public class ContentHasher<T> : IContentHasher
         /// <summary>
         ///     Object pool that holds all instantiated hash algorithms for each hash type.
         /// </summary>
-        private readonly Pool<HashAlgorithm> _algorithmsPool = new Pool<HashAlgorithm>(() => new T());
+        /// <remarks>
+        ///     Cap the number of idle reserve instances in the pool so as to not unnecessarily hold large amounts of memory
+        /// </remarks>
+        private readonly Pool<HashAlgorithm> _algorithmsPool = new Pool<HashAlgorithm>(() => new T(), maxReserveInstances: 10);

```

Comment:
Such a low number could have a perf impact in the cache codebase.

CASaaS will definitely have more then 10 instances, and this logic can unpredictably increase churn.

I definitely agree with the premise of this change, but I'm not sure that this should be done unconditionally.

An alternative solution may be to use 10 from bxl but still use the default value from casaas.